src/design-pages/idoverride_redesign.rst

```diff
@@ -0,0 +1,400 @@
+* `IPA in trust. Client not seeing user with override set on AD user`_
+* `User information is being removed from SSSD local Override database`_
+
+.. _Infopipe ListByCertificate does not return users if more than 1 override matches: https://github.com/SSSD/sssd/issues/4456
+.. _[RFE] Need an option for sss_override so that uidnumber can be added locally even if it is not available on LDAP server: https://bugzilla.redhat.com/show_bug.cgi?id=1896728
+.. _Unexpected interaction between overriding the primary group and the 'auto_private_groups = true' option: https://bugzilla.redhat.com/show_bug.cgi?id=1921315
+.. _RFE\: Permanently Persistent Local (on-system) SSS Overrides both automating importation (such as after the SSS DB is cleared and SSSD restarted) and a standard location for the override export files: https://bugzilla.redhat.com/show_bug.cgi?id=1939527
+.. _IPA in trust. Client not seeing user with override set on AD user: https://bugzilla.redhat.com/show_bug.cgi?id=2010512
+.. _User information is being removed from SSSD local Override database: https://bugzilla.redhat.com/show_bug.cgi?id=1452883
+
+
+Problem statement
+*****************
+The original design for id-overrides is described in
+https://www.freeipa.org/page/V4/Migrating_existing_environments_to_Trust and
+all the features described there should of course work as expected after the
+redesign as well. Only the implementation should be updated with performance
+and reliability in mind.
+
+Initial the id-overrides were planned as a feature for the IPA provider. Some
+time later the local id-overrides were added keeping the way how id-overrides
+were handled by the ipa provider. This meant that when the id-override was
+added the related user was looked up as well and added to the cache together
+with the id-override data. And if the user was removed from the cache for
+whatever reasons the matching id-override was removed from the cache as well.
+This works well for well for IPA since there is a central storage for all
+current id-overrides on the IPA servers. But the local id-override data is
+lost and if the user is added to the cache again there will be no id-override
+data to apply. So it would make sense to decouple the id-override data in
+SSSD's cache form the actual user and group objects.
+
+The id-override data is typically stored in a separate tree/container in the
+cache. There is an exception for the "Default Trust View" on IPA servers where
+the id-override data is directly applied to the user or group objects in the
+cache (see original design for the reason) but with the new design it should be
+possibel to get rid of this special handling. So instead of removing the whole
+id-override object automatically only the back-reference to the user or group
+object can be removed and the id-override object is only removed if the source
+object is removed or the local id-override is explicitly removed.
```

Comment:
I would split this sentence in two or three. Let me try to give you an example but please feel free to do it in another way:

So instead of removing the whole id-override object automatically. only the back-reference to the user or group object is removed. The id-override object is deleted only if the source object is removed or the local id-override is explicitly removed.